modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/baseComponents/BaseAPIEndpointFields.tsx

```diff
@@ -148,6 +179,43 @@ export default function BaseAPIEndpointFields({
 		}
 	}, [data, scopeOptions]);
 
+	useEffect(() => {
+		if (data.httpMethod?.key && httpMethodOptions.length) {
```

Comment:
Here there are things that I don't understand, this useEffect looks "difficult" and that could end in bugs. Let's talk tomorrow to see how can we improve this.

Reply:
Already spoke about this. We will leave it as it is, and will change and divide in subcomponents in the future.